firebase-perf/src/main/java/com/google/firebase/perf/application/FragmentStateMonitor.java

```diff
@@ -17,28 +17,27 @@
 import androidx.annotation.NonNull;
 import androidx.core.app.FrameMetricsAggregator;
 import androidx.fragment.app.Fragment;
-import androidx.fragment.app.FragmentActivity;
 import androidx.fragment.app.FragmentManager;
 import com.google.firebase.perf.logging.AndroidLogger;
+import com.google.firebase.perf.metrics.Trace;
 import com.google.firebase.perf.transport.TransportManager;
 import com.google.firebase.perf.util.Clock;
 import com.google.firebase.perf.util.Constants;
+import java.util.WeakHashMap;
 
 public class FragmentStateMonitor extends FragmentManager.FragmentLifecycleCallbacks {
   private static final AndroidLogger logger = AndroidLogger.getInstance();
-  private final FragmentActivity activity;
+  private final WeakHashMap<Fragment, Trace> fragmentToTraceMap = new WeakHashMap<>();
```

Comment:
Having a map is a good idea. Can you validate few scenarios for this map when there are recycler views in visibility which can cause more fragments to be created. Are we cleaning up the hashmap at such scale is worth accounting for.

Especially scenarios where the recycler view has fragments of which some of them are in visibility and some of them are not (but already created and ready).

Reply:
androidx ViewPager2 is using Fragments inside of a RecyclerView. If needed, I think we should use that to test this behavior.

Reply:
In theory, every activity will have one fragmentStateMonitor, which will hold all the fragments in the map. The map will be cleaned up after transitioning to another activity. Therefore, I think the map size will not be very large.

Reply:
@visumickey I feel like this is a non-blocking investigation that can be parallelized: I want to create a new buganizer to track this investigation, then just finish the current PR with unit tests, then come back to investigate. Would that be okay?

Reply:
I agree, it is a non-blocking investigation and we can always course correct if needed. For now, lets unblock ourselves and move ahead on this.

Reply:
created b/226189464 to tracking this